feat: Cloud Gateway terminology updates (#7093)
Kong/docs.konghq.com: app/konnect/index.md

management platform designed from the ground up for the cloud native era
and delivered as a service. This platform lets you build modern applications
better, faster, and more securely. The control plane is hosted

> in the cloud by Kong, while the data plane is managed by you or Kong in your

Reviewer: Same as in the doc above. I don't think its powerful to say managed by you 'or kong'. I think highlighting choice and flexibility is the most important part.

collaborate and manage their own set of data plane nodes and services without
the risk of impacting other teams and projects. Gateway Manager instantly
provisions a hosted {{site.base_gateway}} control plane and supports securely

> attaching data plane nodes from your fully-managed cloud, or self-managed cloud and hybrid environments.

Reviewer: ```suggestion
attaching self-hosted, or fully-managed data plane nodes and hybrid environments.
```

Reviewer 2: Suggestion off of Angel's suggestion, as the "hybrid environments" portion of the sentence wasn't in the right place.

```suggestion
attaching self-hosted nodes running in cloud or hybrid environments, or using fully-managed data plane nodes.
```
I'm still not 100% on this phrasing, but the point I'm trying to get to it - self-hosted is what runs anywhere, fully-managed doesn't have that control.
